partitura/utils/music.py

```diff
@@ -2871,6 +2871,116 @@ def get_matched_notes(spart_note_array, ppart_note_array, alignment):
     return np.array(matched_idxs)
 
 
+def slice_ppart_by_time(
+    ppart, 
+    start_time: float, 
+    end_time: float,
+    clip_note_off: bool=True,
+    reindex_notes: bool=True):
+    """
+    Get a slice of a PeformedPart by time
+
+    Parameters
+    ----------
+    ppart : `PerformedPart` object
+    start_time : float
+        Starting time in seconds 
+    end_time : float
+        End time in seconds
+    clip_note_off : bool
+        Clip note_off time to end_time
+    reindex_notes : bool
+        Reindex notes in slice starting from n0
+
+    Returns
+    -------
+    ppart_slice :  `PerformedPart` object 
+        A copy of input ppart containing notes, programs and control information 
+        only between `start_time` and `end_time` of ppart
+    """
+    from partitura.performance import PerformedPart
+    from copy import deepcopy
+
+    if not isinstance(ppart, PerformedPart):
+        raise ValueError("Input is not an instance of PerformedPart!")
+
+    assert(start_time < end_time), "Start time not less than end time!"
+
+    ppart_slice = deepcopy(ppart)
```

Comment:
A deepcopy of a larger object is done here to be trimmed down afterward, the slice can be built from scratch.

Reply:
Thank you for pointing this out, the deepcopy is indeed unnecessary. I will re-write this to start from a new class instance.